src/term.rs

```diff
@@ -713,4 +739,11 @@ mod tests {
         assert!(app(abs(Var(1)), Var(1)).is_isomorphic_to(&app(abs(Var(1)), Var(1))));
         assert!(!app(abs(Var(1)), Var(1)).is_isomorphic_to(&app(Var(2), abs(Var(1)))));
     }
+
+    #[test]
+    fn has_free_variables() {
```

Comment:
it would be cool if we had one or more extra cases that are a bit more complex

Reply:
I've added a few more tests in the most recent commit.